src/components/two-factor-authentication/HardwareTokenWorkflow.vue

```diff
@@ -0,0 +1,149 @@
+<script setup lang="ts">
+  import { useDisplay } from 'vuetify';
+  import FormRow from '@/components/form/FormRow.vue';
+  import { ref, type Ref } from 'vue';
+  import { usePersonStore, type Personendatensatz, type PersonStore } from '@/stores/PersonStore';
+  import axios from 'axios';
+  import { useI18n, type Composer } from 'vue-i18n';
+
+  const { t }: Composer = useI18n({ useScope: 'global' });
+  const { mdAndDown }: { mdAndDown: Ref<boolean> } = useDisplay();
+  const dialogText: Ref<string> = ref('');
+  const proceeded: Ref<boolean> = ref(false);
+  const errorThrown: Ref<boolean> = ref(false);
+  const personStore: PersonStore = usePersonStore();
+  // eslint-disable-next-line @typescript-eslint/typedef
+  const emits = defineEmits(['updateHeader', 'onCloseClicked']);
+  emits('updateHeader', 'Hardware-Token zuordnen');
+
+  const serial: Ref<string> = ref('');
+  const otp: Ref<string> = ref('');
+
+  type Props = {
+    errorCode: string;
+    person: Personendatensatz;
+  };
+  const props: Props = defineProps<Props>();
+
+  function handleApiError(error: unknown): void {
+    errorThrown.value = true;
+    if (axios.isAxiosError(error)) {
+      if (error.response && error.response.data) {
+        const message: string = error.response.data.message || 'An unexpected error occurred.';
+        dialogText.value = message;
+      } else {
+        dialogText.value = 'An unexpected error occurred.';
+      }
+    }
+  }
+
+  function cancelCheck(): void {
+    if (errorThrown.value) {
+      serial.value = '';
+      otp.value = '';
+      dialogText.value = '';
+      proceeded.value = false;
+      errorThrown.value = false;
+      emits('updateHeader', 'Hardware-Token zuordnen');
+    } else {
+      emits('onCloseClicked');
+    }
+  }
+
+  async function assignHardwareToken(): Promise<void> {
+    const referrer: string | null = props.person.person.referrer;
+    if (!referrer) return;
+    try {
+      await personStore.assignHardwareToken(referrer, serial.value, otp.value);
+      dialogText.value = t('admin.person.twoFactorAuthentication.hardwareTokenSetUpSuccess');
+    } catch (error) {
+      emits('updateHeader', t('admin.person.twoFactorAuthentication.hardwareTokenSetUpFailure'));
+      handleApiError(error);
+    } finally {
+      proceeded.value = true;
+    }
+  }
+</script>
+
+<template v-slot:activator="{ props }">
+  <v-container v-if="!proceeded">
+    <v-col md="10">
+      <FormRow
+        :errorLabel="''"
+        labelForId="serial-input"
+        :isRequired="true"
+        :label="'Seriennummer'"
```

Comment:
Im Scope vom Ticket steht, dass die beiden Fields validiert werden müssen. Hier muss eine v-form drin sein um die Validierung von Fields zu ermöglichen mit vee-validate. 

Beispiel dafür sind z.B. RolleCreationView.vue ... Du kannst die fields dann auch umbennen in "selectedSeriennummer"" und "selectedOtp", die beiden als Required einstellen und auf die gleiche Art und Weise validieren wie z.B. bei Rollenname usw...

Reply:
V-Form jetzt drin.